lightning/src/ln/channelmanager.rs

```diff
@@ -2555,6 +2590,170 @@ where
+		}
+
+		let chan = channel_lock.get_channel();
+
+		let original_funding_txo = chan.context.get_original_funding_txo().unwrap();
+		let monitor_update = chan.set_funding_outpoint(funding_outpoint, channel_value_satoshis, own_balance, true, &self.logger);
+		if let Some(monitor_update) = &monitor_update {
+			let update_res = self.chain_monitor.update_channel(original_funding_txo, monitor_update);
+			if ChannelMonitorUpdateStatus::Completed != update_res {
+				return Err(APIError::APIMisuseError { err: "Could not update channel funding outpoint.".to_string() });
+			}
+		}
+
+		if ChannelMonitorUpdateStatus::Completed != self.chain_monitor.update_channel_funding_txo(chan.context.get_original_funding_txo().unwrap(), *funding_outpoint, channel_value_satoshis) {
+			return Err(APIError::APIMisuseError { err: "Could not update channel funding transaction.".to_string() });
+		}
+
+		let res = chan.monitor_updating_restored(&self.logger, &self.node_signer, self.genesis_hash, &self.default_configuration, self.best_block.read().unwrap().height());
+
+
+		return Ok(res.commitment_update.unwrap().commitment_signed)
+	}
+
+	fn on_commitment_signed_get_raa_internal(&self, channel_lock: &mut ChannelLock<SP>, commitment_signature: &secp256k1::ecdsa::Signature, htlc_signatures: &[secp256k1::ecdsa::Signature]) -> Result<msgs::RevokeAndACK, APIError> {
+		let commitment_signed = msgs::CommitmentSigned {
+			channel_id: channel_lock.channel.context.channel_id(),
+			signature: *commitment_signature,
+			htlc_signatures: htlc_signatures.to_vec(),
+		};
+
+		let chan = channel_lock.get_channel();
+
+		let original_funding_txo = chan.context.get_original_funding_txo().unwrap();
+		let monitor_update =
+		match chan.commitment_signed(&commitment_signed, &self.logger) {
+			Err(e) => return Err(APIError::APIMisuseError { err: format!("Invalid commitment signed: {:?}", e) }),
+			Ok(res) => res
+		};
+
+		if let Some(monitor_update) = &monitor_update {
+			if ChannelMonitorUpdateStatus::Completed != self.chain_monitor.update_channel(original_funding_txo, monitor_update) {
+				return Err(APIError::APIMisuseError { err: "Could not update channel".to_string() });
+			}
+		}
+
+		let updates = chan.monitor_updating_restored(&self.logger, &self.node_signer, self.genesis_hash, &self.default_configuration, self.best_block.read().unwrap().height());
+
+		Ok(updates.raa.unwrap())
+	}
+
+	fn revoke_and_ack_commitment_internal(&self, channel_lock: &mut ChannelLock<SP>, revoke_and_ack: &RevokeAndACK) -> Result<(), APIError> {
+		let chan = channel_lock.get_channel();
+
+		let original_funding_txo = chan.context.get_original_funding_txo().unwrap();
+
+		// TODO(holzeis): check if we have to consider hold_mon_update here.
+		let updates = chan.revoke_and_ack(revoke_and_ack, &self.fee_estimator, &self.logger, false).map_err(|e| APIError::APIMisuseError { err: format!("{:?}", e) })?;
```

Comment:
@luckysori did you have a look at that flag, I am not sure if false is the right value to set here.

Reply:
I propose https://github.com/p2pderivatives/rust-lightning/pull/15/commits/54e4706d109353bdf02a7f9d4fd2c31b956f6d02. What do you think?

Reply:
looks reasonable to me 👍

Reply:
It was causing a deadlock D:

I've gone for 381c0674 now.